fix: date time picker onChange handler is triggered every time defaultValue changes

frisonisland opened the pull request:
The onChange handler should be triggered only when the user picks a different date.

Ref:
 - CDS-325
 - CO-2006

giuliano176 committed:
test(datetimePicker.test.tsx): add test for onChange invocation

Add a test to verify that the onChange prop is called once when a new
value is set and confirmed in the DateTimePicker component.

frisonisland committed:
fix: add state to manage changes in default value

frisonisland committed:
chore: update action type enums

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zextras/carbonio-design-system?pullRequest=503) <br/>All committers have signed the CLA.

frisonisland committed:
test: print object in test

frisonisland committed:
test: update test descriptions on DateTimePicker.test.tsx

CataldoMazzilli commented on src/components/inputs/datetimePicker/DateTimePicker.tsx:
```diff
@@ -148,7 +152,7 @@ const DateTimePicker = React.forwardRef<DatePicker, DateTimePickerProps>(functio
 
 	useEffect(() => {
-		updateDateTimeState({ type: 'SAVE_AND_UPDATE', value: defaultValue });
+		updateDateTimeState({ type: 'DEFAULT_VALUE_CHANGED', value: defaultValue });
```
If updateDateTimeState dependency is not a triggering dependency, it could be simplified like this and the updateDateTimeState dependency can be removed along with the new case added in the above part.
This could avoid further unwanted rerenders
```suggestion
		dateTimeRef.current = newValue;
		setDateTime(newValue);
```

giuliano176 committed:
chore(DateTimePicker.tsx): remove unused case and update effect

Removed the 'DEFAULT_VALUE_CHANGED' case from the updateDateTimeState
function as it was no longer used. Updated the useEffect hook to directly
set the defaultValue to dateTimeRef and setDateTime.
